stackslib/src/net/p2p.rs

```diff
@@ -3606,12 +3606,7 @@ impl PeerNetwork {
 
             // in Nakamoto epoch, but we might still be doing epoch 2.x things since Nakamoto does
             // not begin on a reward cycle boundary.
-            if cur_epoch.epoch_id == StacksEpochId::Epoch30
-                && (self.burnchain_tip.block_height
-                    <= cur_epoch.start_height
-                        + u64::from(self.burnchain.pox_constants.reward_cycle_length)
-                    || self.connection_opts.force_nakamoto_epoch_transition)
-            {
+            if cur_epoch.epoch_id >= StacksEpochId::Epoch30 && !self.stacks_tip.is_nakamoto {
```

Comment:
Can you update the comment above this line to say something like `In nakamoto, but our stacks tip is still an epoch 2.x block, so we may need to download more 2.x blocks before cutting over entirely to nakamoto`?

Reply:
Sure